### PR TITLE
feat(oauth): uplift fxa-oauth-server directory, update CI setups

### PR DESCRIPTION
**Pull request by vladikoff**
@rfk suggested that first step should probably be just moving the files. 

I did an exploration of actually merging the servers in https://github.com/mozilla/fxa-auth-server/pull/2669 but we should do this part first.

### CODE REVIEW

File: fxa-oauth-server/lib/routes/root.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    if (commitHash) {
+      return sendReply();
+    }
+
+    function runGitCmd() {
+      return new Promise((resolve) => {
+        // figure it out from .git
+        var gitDir = path.resolve(__dirname, '..', '..', '..', '.git');
```
Comment (vladikoff) [inline fxa-oauth-server/lib/routes/root.js]: This is the only code change besides the CI files

**Comment by philbooth** [comment]
> ...I would like to hear from others before we click the big button.

Not sure if this includes me but fwiw I am 👍 to the concept and 👍 to the directory structure used in this PR and therefore 👍 in general even though I haven't actually code-reviewed the 24k lines.

File: .circleci/config.yml [hunk 2]
```diff
@@ -66,6 +66,76 @@ jobs:
...
+            if [[ "${CIRCLE_BRANCH}" == feature* ]] || [[ "${CIRCLE_BRANCH}" == dockerpush* ]]; then
+              DOCKER_TAG="${CIRCLE_BRANCH}"
+            fi
+
+            if [ -n "${CIRCLE_TAG}" ]; then
+              DOCKER_TAG="$CIRCLE_TAG"
+            fi
+
+            if [ -n "${DOCKER_TAG}" ]; then
+              echo "$DOCKER_OAUTH_PASS" | docker login -u "$DOCKER_OAUTH_USER" --password-stdin
```
[the remark this answers is not in the document]
Comment (jbuck) [reply .circleci/config.yml]: I'm wondering how the tagging is gonna work here.. I guess oauth & auth will share the same tags?

[the remark this answers is not in the document]
Comment (vladikoff) [reply .circleci/config.yml]: @jbuck yeap

[the remark this answers is not in the document]
Comment (jbuck) [reply .circleci/config.yml]: Okay, done

**Comment by jbuck** [comment]
This can be merged now

**Comment by rfk** [comment]
I was pretty much ready to click the "merge" button on this, then I realized one thing we'd lose in doing so: the commit history in fxa-oauth-server.  That's a fairly rich set of historical data, and while it would still be available in the old fxa-oauth-server repo, we'd lose the ability to easily do things like `git blame` and `git bisect` on the oauth-server files.

I'm going to do a quick experiment to see if I can replicate this PR while keeping the fxa-oauth-server history intact.  It may or may not be worthwhile in practice, but it'll be worth knowing either way.

**Comment by rfk** [comment]
I have an alternative version of this in https://github.com/mozilla/fxa-auth-server/pull/2680 that preserves the commit history.  It's otherwise identical except for the inclusion of the recent https://github.com/mozilla/fxa-oauth-server/pull/618.  Using `git log` and `git blame` and so-on seem to operate sensibly, but I'd like to hear what others think about the benefits (and downsides!) of bringing across the history.

**Comment by rfk** [comment]
We merged the history-preserving version of this in #2680

**Pull request closed without merging**